testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -213,6 +223,25 @@ impl BlockMinerThread {
                         warn!("Failed to propose block to stackerdb: {e:?}");
                     }
                 }
+                let mut chain_state = neon_node::open_chainstate_with_faults(&self.config)
+                    .expect("FATAL: could not open chainstate DB");
+                if let Ok(aggregate_public_key) = NakamotoChainState::get_aggregate_public_key(
+                    &mut chain_state,
+                    &sort_db,
+                    &sort_db.index_handle_at_tip(),
+                    &new_block,
+                ) {
+                    if let Some(signature) = self.wait_for_signature(
+                        &stackerdbs,
+                        &aggregate_public_key,
+                        &signer_signature_hash,
+                    ) {
+                        // TODO remove self signing once casting aggregate vote is done
+                        new_block.header.signer_signature = signature;
+                    } else {
+                        debug!("Failed to get a signed block from signers");
+                    }
+                };
```

Comment:
I think this behavior should be mutually exclusive with `self.config.self_signing()` --

If `self_signing()` is enabled, then it shouldn't use (or wait for) the signer-set signature.
If `self_signing()` is disabled, then it should use the signer-set signature and then broadcast the block.

I'd recommend moving this functionality into a new function like `.wait_for_signer_signature_and_broadcast`, and then that could be invoked at line 252, replacing the current warning when self-signing is disabled.